storage/series/buffer.go

```diff
@@ -236,21 +236,28 @@ func (b *dbBuffer) DrainAndReset() drainAndResetResult {
 
 func bucketDrainAndReset(now time.Time, b *dbBuffer, idx int, start time.Time) int {
 	mergedOutOfOrderBlocks := 0
+
 	if b.buckets[idx].needsDrain(now, start) {
-		merged := b.buckets[idx].discardMerged()
-		if merged.merges > 0 {
-			mergedOutOfOrderBlocks = 1
-		}
-		if merged.block.Len() > 0 {
-			// If this block was read mark it as such
-			if lastRead := b.buckets[idx].lastRead(); !lastRead.IsZero() {
-				merged.block.SetLastReadTime(lastRead)
-			}
-			b.drainFn(merged.block)
-		} else {
+		// Rotate the buffer to a block, merging if required
+		result, err := b.buckets[idx].discardMerged()
+		if err != nil {
 			log := b.opts.InstrumentOptions().Logger()
-			log.Errorf("buffer drain tried to drain empty stream for bucket: %v",
-				start.String())
+			log.Errorf("buffer merge encode error: %v", err)
+		} else {
+			if result.merges > 0 {
+				mergedOutOfOrderBlocks++
+			}
+			if result.block.Len() < 1 {
```

Comment:
could this be == 0? Are you just protecting against accidental negative values?

I think `if !result.block.Len() > 0` might be clearer, `< 1` is jarring when talking about a length

Reply:
Ha, yeah its the negative thing - a habit I picked up a while ago defending against values passed by users.  I'm not sure how Len() is implemented under the hood for a block when I was writing the code I think.